File: fintechplatform-ui/src/main/java/com/fintechplatform/android/transactions/ui/itemview/TransactionItemPresenterModule.java
package com.fintechplatform.android.transactions.ui.itemview;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class TransactionItemPresenterModule {
    private final TransactionItemContract.View view;

    public TransactionItemPresenterModule(TransactionItemContract.View view) {
        this.view = view;
    }

    @Provides
    @Singleton
    TransactionItemContract.View provideTransactionItemContractView() {
        return view;
    }

    @Provides
    @Singleton
    TransactionItemContract.Presenter provideTransactionItemPresenter() {
        return new TransactionItemPresenter(view);
    }
}

File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/transactions/di/TransactionItemPresenterModule.kt
package com.fintechplatform.ui.transactions.di

import com.fintechplatform.ui.transactions.ui.itemview.TransactionItemContract
import com.fintechplatform.ui.transactions.ui.itemview.TransactionItemPresenter
import dagger.Module
import dagger.Provides
import javax.inject.Singleton


@Module
class TransactionItemPresenterModule(private val view: TransactionItemContract.View) {

    @Provides
    @Singleton
    internal fun provideTransactionItemContractView(): TransactionItemContract.View {
        return view
    }

    @Provides
    @Singleton
    internal fun provideTransactionItemPresenter(): TransactionItemContract.Presenter {
        return TransactionItemPresenter(view)
    }
}
